### PR TITLE
update to 0.981

### PR DESCRIPTION
**Pull request by katietz**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -49,6 +50,7 @@ test:
     - stubtest --help
 
 about:
+  # https is not supported by mype-lang.org server from within our build-net
   home: http://www.mypy-lang.org/
   license: MIT
   license_family: MIT
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:56]: doc_url should be with HTTPS

File: recipe/meta.yaml [hunk 2]
```diff
@@ -20,16 +20,17 @@ build:
...
   host:
-    - python >=3.6
+    - python
     - pip
     - setuptools
     - wheel
   run:
-    - python >=3.6
+    - python
     - mypy_extensions >=0.4.3,<0.5.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Should be `mypy_extensions >=0.4.3`, https://github.com/python/mypy/blob/c7b4714e1f5e3cb8f3fec7426b6538fe1a3dcab1/setup.py#L215

File: recipe/meta.yaml [hunk 3]
```diff
@@ -20,16 +20,17 @@ build:
 
 requirements:
   host:
-    - python >=3.6
+    - python
     - pip
     - setuptools
     - wheel
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:26]: The upstream source https://github.com/python/mypy/blob/c7b4714e1f5e3cb8f3fec7426b6538fe1a3dcab1/pyproject.toml#L6-L16 and conda-forge https://github.com/conda-forge/mypy-feedstock/blob/e0d9f90b3a924243fa31522e9e144aaa40797a93/recipe/meta.yaml#L55-L61 are using runtime and other dependencies in `host` too. I'm not sure if they are really needed in host

**Comment by mcg1969** [comment]
The mypy home page suggests 0.990 is out (a single version bump). Any reason in particular we shouldn't upgrade to that?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -20,16 +20,17 @@ build:
 
 requirements:
   host:
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:22]: It looks like the source code has **mypyc** C extension and it requires the C compiler, https://github.com/conda-forge/mypy-feedstock/commit/f9824ddb36ab67b6d6a2baab7034ff53249821a4#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR35. But **mypyc** will be moved to another repo, see https://github.com/python/mypy/tree/v0.981/mypyc#mypyc-mypy-to-python-c-extension-compiler.
Or we can skip building it?

File: recipe/meta.yaml (hunk 2 above)
Comment (katietz) [reply recipe/meta.yaml]: well. actually by looking to it, it should be >=0.4.3,<2

File: recipe/meta.yaml (hunk 3 above)
Comment (katietz) [reply recipe/meta.yaml:26]: no, they are not

File: recipe/meta.yaml (hunk 2 above)
Comment (katietz) [reply recipe/meta.yaml]: err, ...<1

File: recipe/meta.yaml (hunk 4 above)
Comment (katietz) [reply recipe/meta.yaml:22]: we don't need this

**Review by skupr-anaconda — APPROVED** [review]

**Review by sumit0190 — APPROVED** [review]

**Pull request merged**